### PR TITLE
Feature/3402 drawer layout example

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3402

#### Changes proposed in this Pull Request:
- Added Drawer Layout examples

#### Screenshots:
Screenshot 2022-09-19 at 2 09 40 PM

#### To Test:
- yarn start:reactdev
**Note: Need to manually build the react components and paste node modules in docs**
There is an issue with temporary drawer positioning for which issue has been logged https://github.com/brightlayer-ui/react-component-library/issues/534

### CODE REVIEW

**Comment by emclaug2** [comment]
This isn't how the rail variant should be appearing - do you know why it's off-centered? 
image

**Comment by emclaug2** [comment]
No skinny rail variant example?

**Comment by bkarambe** [comment]
@emclaug2 Please review.

File: docs/src/componentDocs/DrawerLayout/examples/CondensedRailDrawerExample.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { Devices, LocationOn } from '@mui/icons-material';
+
+export const CondensedRailDrawerExample = (): JSX.Element => (
+    <DrawerLayout
+        drawer={
+            <Drawer open={true} width={332} variant={'rail'} condensed={true}>
+                <DrawerBody>
+                    <DrawerNavGroup>
+                        <DrawerRailItem
+                            sx={{ width: '100%', cursor: 'pointer' }}
```
[the remark this answers is not in the document]
Comment (bkarambe) [reply docs/src/componentDocs/DrawerLayout/examples/CondensedRailDrawerExample.tsx:14]: Logged an issue and removed sx https://github.com/brightlayer-ui/react-component-library/issues/541

**Review by huayunh — APPROVED** [review]
Thank you. I love how we are able to discover so many issues with our component library.

**Comment by ektasawant** [comment]
@bkarambe , can you fix conflicts and merge this PR?

**Pull request merged**